proof/proof_test.go

```diff
@@ -505,6 +507,93 @@ func TestGenesisProofVerification(t *testing.T) {
 	test.WriteTestVectors(t, generatedTestVectorName, testVectors)
 }
 
+// TestUpdateManualProofs is only executed when updating test vectors and is
+// responsible for updating the following files:
+//   - testdata/proof_tlv_encoding_other.json
+//   - testdata/proof.hex
+//   - testdata/proof-file.hex
+func TestUpdateManualProofs(t *testing.T) {
+	if !test.GeneratingTestVectors {
+		t.Skip("Not generating test vectors, skipping proof file " +
+			"updates")
+	}
+
+	// First, we update the proof_tlv_encoding_other.json file. This file
+	// contains actual proofs from a random integration test run. We know
+	// that the proof's content is valid, but in case the encoding of it
+	// changes, we just re-encode it here and update the file.
+	var (
+		otherVectors = &TestVectors{}
+		fileProofs   []Proof
+	)
+
+	test.ParseTestVectors(t, otherTestVectorName, &otherVectors)
+	for idx := range otherVectors.ValidTestCases {
+		proof := otherVectors.ValidTestCases[idx].Proof.ToProof(t)
+
+		// We need to update the inclusion proofs in case anything in
+		// the asset serialization changed.
+		assetTree, err := commitment.NewAssetCommitment(&proof.Asset)
+		require.NoError(t, err)
+		assetTapTree, err := commitment.NewTapCommitment(assetTree)
+		require.NoError(t, err)
+		_, commitmentProof, err := assetTapTree.Proof(
+			proof.Asset.TapCommitmentKey(),
+			proof.Asset.AssetCommitmentKey(),
+		)
+		require.NoError(t, err)
+		proof.InclusionProof.CommitmentProof.Proof = *commitmentProof
+
+		var buf bytes.Buffer
+		require.NoError(t, proof.Encode(&buf))
+
+		expectedStr := hex.EncodeToString(buf.Bytes())
+		otherVectors.ValidTestCases[idx].Expected = expectedStr
+
+		comment := otherVectors.ValidTestCases[idx].Comment
+		switch {
+		// The proof with the comment "valid regtest split proof" is
+		// also the proof we encode in the proof.hex file.
+		case comment == "valid regtest split proof":
+			err := os.WriteFile(
+				proofHexFileName, []byte(expectedStr), 0644,
+			)
+			require.NoError(t, err)
+
+		// The proof with the comment "valid regtest ownership proof" is
+		// also the proof we encode in the ownership-proof.hex file.
+		case comment == "valid regtest ownership proof":
+			err := os.WriteFile(
+				ownershipProofHexFileName, []byte(expectedStr),
+				0644,
+			)
+			require.NoError(t, err)
+
+		// Any proof that starts with the comment "valid regtest proof
+		// file index X", we add to the proof-file.hex file.
+		case strings.Contains(comment, "valid regtest proof file index"):
+			fileProofs = append(fileProofs, *proof)
+		}
```

Comment:
add default case for unexpected comment?

Reply:
There are more than those three test cases in the file, we just export some of them into a different format to test. So we already ignore a bunch of them. Will make it more apparent by adding an empty default case with a comment.